biochatter/vectorstore_host.py

```diff
@@ -1,196 +1,541 @@
-                if not text_field:
-                    continue
-                docname = matched_grp[0]
-                docname = base64_to_string(docname)
-                self._collections.append(
-                    VectorCollection(
-                        col_name=col_name,
-                        doc_name=docname,
-                        text_field=text_field,
-                    )
-                )
-                break
-
-    @property
-    def collections(self) -> List[Dict[str, str]]:
-        return [
-            {
-                "document_name": col.document_name,
-                "collection_name": col.collection_name,
-            }
-            for col in self._collections
+        if embedding_exists:
+            self._load_embeddings_collection()
+        else:
+            self._create_embeddings_collection()
+
+        if meta_exists:
+            self._load_metadata_collection()
+        else:
+            self._create_metadata_collection()
+
+        self._create_metadata_collection_index()
+        self._col_metadata.load()
+
+    def _load_embeddings_collection(self) -> None:
+        """
+        Load embeddings collection from currently active database.
+        """
+        try:
+            self._col_embeddings = Milvus(
+                embedding_function=self._embedding_func,
+                collection_name=self._embedding_name,
+                connection_args=self._connection_args,
+            )
+        except MilvusException as e:
+            logger.error(
+                f"Failed to load embeddings collection {self._embedding_name}."
+            )
+            raise e
+
+    def _create_embeddings_collection(self) -> None:
+        """
+        Create embedding collection.
+        All fields: "meta_id", "vector"
+        """
+        try:
+            self._col_embeddings = Milvus(
+                embedding_function=self._embedding_func,
+                collection_name=self._embedding_name,
+                connection_args=self._connection_args,
+            )
+        except MilvusException as e:
+            logger.error(
+                f"Failed to create embeddings collection {self._embedding_name}"
+            )
+            raise e
+
+    def _load_metadata_collection(self) -> None:
+        """
+        Load metadata collection from currently active database.
+        """
+        self._col_metadata = Collection(
+            self._metadata_name,
+            using=self.alias,
+        )
+        self._col_metadata.load()
+
+    def _create_metadata_collection(self) -> None:
+        """
+        Create metadata collection.
+        All fields: "id", "name", "author", "title", "format", "subject", "creator", "producer",
+        "creationDate", "modDate", "source", "embedding", "isDeleted"
+        As vector database requires one vector field, we will create a fake vector "embedding".
+        Field "isDeleted" is used to specify if the document is deleted.
```

Comment:
@fengsh27 I am not 100% sure how the deletion process works. Do we only mark deleted embeddings / documents, meaning they aggregate over time, or do they actually disappear from the DB at some point? The latter would be preferable.